lib/standard/bytes.nit

```diff
@@ -0,0 +1,129 @@
+# Introduces classes and methods to treat byte streams and arrays as such
+module bytes
+
+import kernel
+import collection::array
+intrude import string
+
+# A buffer containing Byte-manipulation facilities
+#
+# TODO: Change the bound to Byte when available in stdlib and bootstrap
+class ByteBuffer
+	super AbstractArray[Int]
+
+	# A NativeString being a char*, it can be used as underlying representation here.
+	private var items: NativeString
```

Comment:
I agree with your point, however I'm not sure how to do it otherwise, ideally the default constructor should be protected or private, but with the actual spec, it needs to be public.

And having the arguments declared as noinit to be initialized with the available named constructors is a bit of a PITA imho since a call to the default constructor does just that in an automatic fashion